py/jpy/src/main/c/jni/org_jpy_PyLib.c

```diff
@@ -2145,6 +2145,21 @@ JNIEXPORT jboolean JNICALL Java_org_jpy_PyLib_hasGil
     return result;
 }
 
+/*
+ * Class:     org_jpy_PyLib
+ * Method:    ensureGil
+ * Signature: (Ljava/util/function/Supplier;)Ljava/lang/Object;
+ */
+JNIEXPORT jobject JNICALL Java_org_jpy_PyLib_ensureGil
+  (JNIEnv* jenv, jclass jLibClass, jobject supplier)
+{
+    jobject result;
+    JPy_BEGIN_GIL_STATE
+    result = (*jenv)->CallObjectMethod(jenv, supplier, JPy_Supplier_get_MID);
+    JPy_END_GIL_STATE
+    return result;
```

Comment:
Do we need exception checking / handling?

Reply:
Probably. I'll look into it, see how to let it propagate. Off the top of my head, just plain java throwables need to be handled? No explicit allocations take place, so we don't need to trigger a java oom/etc, right?

Reply:
See the test added, it doesn't appear to be necessary. There is a JNI func to call from C to check if an exception is set in java, but it doesnt need to be checked if you don't want the exception addressed until you get back into java (at which time it is automatically handled).